api/desecapi/tests/test_throttling_user.py

```diff
@@ -0,0 +1,80 @@
+from unittest import mock
+import time
+
+from django.core.cache import cache
+from django.test import TestCase, override_settings
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.test import APIRequestFactory, force_authenticate
+
+from desecapi.models import User
+
+
+class MockView(APIView):
+    @property
+    def throttle_classes(self):
+        # Need to import here so that the module is only loaded once the settings override is in effect
+        from desecapi.throttling import UserRateThrottle
+
+        return (UserRateThrottle,)
+
+    def get(self, request):
+        return Response("foo")
+
+
+class ThrottlingTestCase(TestCase):
+    """
+    Based on DRF's test_throttling.py.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.factory = APIRequestFactory()
+
+    def _test_requests_are_throttled(self, counts, user=None):
+        def do_test():
```

Comment:
this seems to be only called once so it doesn't need to be encapsulated. "de"capsulating it could improve readability of the code